ical_exporter.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import "io" 
+
+type ICalTaskExporter struct {
+	Version string 
+	PRODID string 
+	CALSCALE string
+	TimeFormat string // UTC time format string (layout string passed to time.Format)
+}
+
+// tryWriteStrings is a helper function used to write strings to a writer if an error has not previously occured
+// 	if an error occurs during the write the "error" argument will receive the error 
+func tryWriteStrings(error *error, writer io.Writer, strings ...string) {
+	if strings != nil {
+		for _, str := range strings {
+			if *error != nil {
+				return
+			}
+			_, *error = io.WriteString(writer, str)
+		}
+	}
+}
+
+func (iCalTaskExporter ICalTaskExporter) getVersion() string {
+	if iCalTaskExporter.Version == "" {
+		return "2.0"
+	}
+	return iCalTaskExporter.Version
+}
+
+func (iCalTaskExporter ICalTaskExporter) getPRODID() string {
+	if iCalTaskExporter.PRODID == "" {
+		return "-//mlimaloureiro/golog"
+	}
+	return iCalTaskExporter.PRODID
+}
+
+func (iCalTaskExporter ICalTaskExporter) getCALSCALE() string {
+	if iCalTaskExporter.CALSCALE == "" {
+		return "GREGORIAN"
+	}
+	return iCalTaskExporter.CALSCALE
+}
+
+func (iCalTaskExporter ICalTaskExporter) getTimeFormat() string {
+	if iCalTaskExporter.TimeFormat == "" {
+		return "20060102T150405Z"
+	}
+	return iCalTaskExporter.TimeFormat
+}
+
+func (iCalTaskExporter ICalTaskExporter) writeICalHeader(writer io.Writer) error {
+	var error error
+
+	tryWriteStrings(&error, writer, "BEGIN:VCALENDAR\n")
+	if error == nil {
```

Comment:
```
var err error
if err == nil {
```

let's use vertical proximity to check where to define the variables

Reply:
also, this is a nice topic. for defining errors, I think we should standardize `err` for the variable to avoid confusion with the `error` keyword. what do you think ? need input here aguiar

Reply:
I thought about using `err` instead of `error` but was afraid it would lose readability.

What do you mean with "vertical proximity"? is it about the new line @ line 55?

Reply:
we should use `err`. if you do like that, you don't have access to error interface (error is an interface) in the function again. 

please refactor to use `err` instead. As Miguel said it's the go standard.